plugins/inputs/t128_tank/t128_tank_test.go

```diff
@@ -240,3 +239,62 @@ func TestIndexParsing(t *testing.T) {
 	_, err = newIndex("foo")
 	assert.Error(t, err)
 }
+
+func newMetric(name string, tags map[string]string, fields map[string]interface{}) telegraf.Metric {
+	if tags == nil {
+		tags = map[string]string{}
+	}
+	if fields == nil {
+		fields = map[string]interface{}{}
+	}
+	m := metric.New(name, tags, fields, time.Now())
+	return m
+}
+
+func TestUnreasonableTimestamp(t *testing.T) {
+	testCases := []struct {
+		Name                  string
+		UnreasonableTimestamp time.Time
+		ExpectedMessage       time.Time
+		Precision             time.Duration
+	}{
+		{
+			Name:            "Precision in Nanosecond",
+			ExpectedMessage: time.Unix(24*60*60, 0),
+			Precision:       time.Nanosecond,
+		},
+		{
+			Name:            "Precision is Empty",
+			ExpectedMessage: time.Unix(0, 0),
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.Name, func(t *testing.T) {
+			fmt.Println(testCase.Name)
+			metric := newMetric("test_metric", nil, nil)
+			metric.SetTime(testCase.UnreasonableTimestamp)
+
+			plugin := &T128Tank{
+				Topic:     "test",
+				Precision: config.Duration(testCase.Precision),
+			}
+			plugin.Init()
+			if plugin.adjustTime == nil {
+				unreasonableTimestamp := time.Unix(0, 0).Add(24 * time.Hour)
+				plugin.adjustTime = func(m telegraf.Metric) {
+					mTime := m.Time()
+					if mTime.Before(unreasonableTimestamp) {
+						adjustedSeconds := unreasonableTimestamp.Unix()
+						m.SetTime(time.Unix(adjustedSeconds, 0))
+					}
+				}
+			}
+			plugin.adjustTime(metric)
+
+			if !metric.Time().Equal(testCase.ExpectedMessage) {
```

Comment:
You set the metric time to `testCase.UnreasonableTimestamp` before any adjusting was done. The code could noop and this test would pass.